lib/sectorbuilder/sectorbuilder.go

```diff
@@ -26,36 +37,120 @@ type SectorInfo = sectorbuilder.SectorInfo
 
 type SealTicket = sectorbuilder.SealTicket
 
-type SealedSectorMetadata = sectorbuilder.SealedSectorMetadata
+type SealSeed = sectorbuilder.SealSeed
+
+type SealPreCommitOutput = sectorbuilder.SealPreCommitOutput
+
+type SealCommitOutput = sectorbuilder.SealCommitOutput
+
+type PublicPieceInfo = sectorbuilder.PublicPieceInfo
+
+type RawSealPreCommitOutput = sectorbuilder.RawSealPreCommitOutput
 
 const CommLen = sectorbuilder.CommitmentBytesLen
 
 type SectorBuilder struct {
 	handle unsafe.Pointer
+	ds dtypes.MetadataDS
+	idLk sync.Mutex
+
+	ssize  uint64
 
 	Miner address.Address
+
+	stagedDir string
+	sealedDir string
+	cacheDir  string
+
+	rateLimit chan struct{}
 }
 
-type SectorBuilderConfig struct {
-	SectorSize  uint64
-	Miner       address.Address
+type Config struct {
+	SectorSize uint64
+	Miner      address.Address
+
+	WorkerThreads uint8
+
+	CacheDir    string
 	SealedDir   string
 	StagedDir   string
 	MetadataDir string
 }
 
-func New(cfg *SectorBuilderConfig) (*SectorBuilder, error) {
+func New(cfg *Config, ds dtypes.MetadataDS, lc fx.Lifecycle) (*SectorBuilder, error) {
```

Comment:
It feels weird to depend on something from fx here, makes it more difficult to construct one of these in a test (even if it is okay to pass nil)